src/smart-escrow/SmartEscrow.sol

```diff
@@ -0,0 +1,190 @@
+        vestingEventTokens = numVestingEventTokens;
+
+        _transferOwnership(escrowOwner);
+    }
+
+    /**
+     * @dev Allow escrow owner (2-of-2 multisig with beneficiary and benefactor
+     * owners as signers) to terminate the contract.
+     *
+     * Emits a {ContractTerminated} event.
+     */
+    function terminate(address returnAddress) external onlyOwner {
+        contractTerminated = true;
+        emit ContractTerminated();
+        withdrawUnvestedTokens(returnAddress);
+    }
+
+    /**
+     * @dev Allow contract owner to update beneficiary owner address.
+     *
+     * Emits a {BeneficiaryOwnerUpdated} event.
+     */
+    function updateBeneficiaryOwner(address newBeneficiaryOwner) external onlyOwner {
+        if (newBeneficiaryOwner == address(0)) revert AddressIsZeroAddress();
+        if (beneficiaryOwner != newBeneficiaryOwner) {
+            address oldBeneficiaryOwner = beneficiaryOwner;
+            beneficiaryOwner = newBeneficiaryOwner;
+            emit BeneficiaryOwnerUpdated(oldBeneficiaryOwner, newBeneficiaryOwner);
+        }
+    }
+
+    /**
+     * @dev Allow beneficiary owner to update beneficiary address.
+     *
+     * Emits a {BeneficiaryUpdated} event.
+     */
+    function updateBeneficiary(address newBeneficiary) external {
+        if (msg.sender != beneficiaryOwner) revert CallerIsNotOwner(msg.sender, beneficiaryOwner);
+        if (newBeneficiary == address(0)) revert AddressIsZeroAddress();
+        if (beneficiary != newBeneficiary) {
+            address oldBeneficiary = beneficiary;
+            beneficiary = newBeneficiary;
+            emit BeneficiaryUpdated(oldBeneficiary, newBeneficiary);
+        }
+    }
+
+    /**
+     * @dev Release OP tokens that have already vested.
+     *
+     * Emits a {OPTransfered} event.
+     */
+    function release() public {
+        if (contractTerminated == true) revert ContractIsTerminated();
+        uint256 amount = releasable();
+        if (amount > 0) {
+            released += amount;
+            emit OPTransfered(amount, beneficiary);
+            SafeERC20.safeTransfer(OP_TOKEN, beneficiary, amount);
+        }
+    }
+
+    /**
+     * @dev Allow withdrawal of remaining tokens to provided address if contract is terminated
+     *
+     * Emits a {OPTransfered} event.
+     */
+    function withdrawUnvestedTokens(address returnAddress) public onlyOwner {
+        if (contractTerminated == false) revert ContractIsNotTerminated();
+        if (returnAddress == address(0)) revert AddressIsZeroAddress();
+        uint256 amount = OP_TOKEN.balanceOf(address(this));
+        if (amount > 0) {
+            emit OPTransfered(amount, returnAddress);
+            SafeERC20.safeTransfer(OP_TOKEN, returnAddress, amount);
+        }
+    }
+
+    /**
+     * @dev Getter for the amount of releasable OP.
+     */
+    function releasable() public view returns (uint256) {
+        return vestedAmount(block.timestamp) - released;
```

Comment:
It's unfortunate we have to SSLOAD `released` twice. Instead, we might be able to change the implementation of `_vestingSchedule` so that it can be used directly by both `releasable` and `vestedAmount` separately.

Reply:
this is an interesting insight! I'm trying to minimize the diff from the OZ Vesting wallet implementation, so am inclined to leave it as is, but let me know if you have a specific proposal

Reply:
Makes sense to me; it's fairly straightforward as is so may only be caught by auditing as a way to reduce gas if someone re-architects this